test/tools/ossfuzz/StackReuseCodegenFuzzer.cpp

```diff
@@ -0,0 +1,146 @@
+
+#include <test/tools/ossfuzz/YulEvmoneInterface.h>
+
+#include <libyul/Exceptions.h>
+
+#include <libyul/backends/evm/EVMCodeTransform.h>
+#include <libyul/backends/evm/EVMDialect.h>
+
+#include <libevmasm/Instruction.h>
+
+#include <liblangutil/EVMVersion.h>
+
+#include <evmone/evmone.h>
+
+#include <src/libfuzzer/libfuzzer_macro.h>
+
+#include <fstream>
+
+using namespace solidity;
+using namespace solidity::test;
+using namespace solidity::test::fuzzer;
+using namespace solidity::yul;
+using namespace solidity::yul::test;
+using namespace solidity::yul::test::yul_fuzzer;
+using namespace solidity::langutil;
+using namespace std;
+
+static evmc::VM evmone = evmc::VM{evmc_create_evmone()};
+
+DEFINE_PROTO_FUZZER(Program const& _input)
+{
+	bool filterStatefulInstructions = true;
+	bool filterUnboundedLoops = true;
+	ProtoConverter converter(
+		filterStatefulInstructions,
+		filterUnboundedLoops
+	);
+	string yul_source = converter.programToString(_input);
+	// Fuzzer also fuzzes the EVM version field.
+	langutil::EVMVersion version = converter.version();
+	EVMHost hostContext(version, evmone);
+	hostContext.reset();
+
+	if (const char* dump_path = getenv("PROTO_FUZZER_DUMP_PATH"))
+	{
+		ofstream of(dump_path);
+		of.write(yul_source.data(), static_cast<streamsize>(yul_source.size()));
+	}
+
+	// Do not proceed with tests that are too large. 1200 is an arbitrary
+	// threshold.
+	if (yul_source.size() > 1200)
+		return;
+
+	YulStringRepository::reset();
+
+	solidity::frontend::OptimiserSettings settings = solidity::frontend::OptimiserSettings::full();
+	settings.runYulOptimiser = false;
+	settings.optimizeStackAllocation = false;
+	bytes unoptimisedByteCode;
+	try
+	{
+		unoptimisedByteCode = YulAssembler{version, settings, yul_source}.assemble();
+	}
+	catch (solidity::yul::StackTooDeepError const&)
+	{
+		return;
+	}
+
+	evmc::result deployResult = YulEvmoneUtility{}.deployCode(unoptimisedByteCode, hostContext);
+	if (deployResult.status_code != EVMC_SUCCESS)
+		return;
+	auto callMessage = YulEvmoneUtility{}.callMessage(deployResult.create_address);
+	evmc::result callResult = hostContext.call(callMessage);
+	// If the fuzzer synthesized input does not contain the revert opcode which
+	// we lazily check by string find, the EVM call should not revert.
+	bool noRevertInSource = yul_source.find("revert") == string::npos;
+	if (noRevertInSource)
+		solAssert(
+			callResult.status_code != EVMC_REVERT,
+			"SolidityEvmoneInterface: EVM One reverted"
+		);
+	// Out of gas errors are problematic because it is possible that the
+	// optimizer makes them go away, making EVM state impossible to
+	// compare in general.
+	if (callResult.status_code == EVMC_OUT_OF_GAS)
+		return;
+
+	if (YulEvmoneUtility{}.checkSelfDestructs(hostContext, deployResult.create_address))
+		return;
+	ostringstream unoptimizedStorage;
+	hostContext.print_storage_at(deployResult.create_address, unoptimizedStorage);
+
+	settings.runYulOptimiser = true;
```

Comment:
Do you think it makes sense to have the fuzzer set this partly to true, partly to false? Both cases are interesting... maybe with a strong bias towards ``true``... Like a random number between 0 and 10 and set this to ``number > 0`` or sth like that?
I think the issue with the delayed return variable PR wouldn't have been hit, because the yul optimizer would have flattened/hoisted the functions in the input...